tests/tls_handler_test.c

```diff
@@ -391,19 +385,16 @@ static int s_tls_local_server_tester_init(
     int server_index,
     const char *cert_path,
     const char *pkey_path) {
+    (void)server_index;
```

Comment:
just remove `server_index` entirely from this function call

It was [added here](https://github.com/awslabs/aws-c-io/commit/6ef51b4b48591e61d92215f158e359a5becaddff#diff-f5e47051107fb5de8f5f6bd7b426bcc0898aab4a1695334ffcdd1e552aaf0769R376). Seems like it was just trying to fix a test that made 2 servers and they got the same socket name. But that's not a problem anymore